### DISCUSSION TITLE
Localation for Quasar

### DISCUSSION THREAD

**Discussion by tolybay63** [quasarframework/quasar#12663] [General - Components / Directives / etc]
Hello!
Please add Kazakh language localization (kz.js)

Sincerely, Tolybai Kuanov

**Comment by MilosPaunovic** [comment]
Quasar heavily relies on community support for localization, so if you're willing to provide a PR, I'm sure the guys maintainig the framework would be more than happy to merge it.

You can take a look at this folder:
[https://github.com/quasarframework/quasar/tree/dev/ui/lang](https://github.com/quasarframework/quasar/tree/dev/ui/lang)
which containt all of the translation files.

If you would like to add new, use:
[https://github.com/quasarframework/quasar/blob/dev/ui/lang/en-US.js]( https://github.com/quasarframework/quasar/blob/dev/ui/lang/en-US.js)
or any other language pack as a base and translate it to Kazakh.

**Reply by tolybay63** [reply]
Thank you!

**Comment by o-r-e** [comment]
Hello!

May be name of language pack file must be "kk.js"? I saw language code "kk" at https://en.wikipedia.org/wiki/Kazakh_language, in right panel, in "Language codes" section.

Also I tried to test locales in few browsers by running next javascript code in browser console:

```js
(() => {
  console.clear();
  const date = new Date(2020, 2, 22, 15, 00, 00, 000);
  const languageNames = new Intl.DisplayNames(['en'], { type: 'language' });
  ['kz', 'kk'].forEach((locale) => {
    const languageName = languageNames.of(locale);
    const f = new Intl.DateTimeFormat(locale, { dateStyle: 'full', timeStyle: 'full' });
    const text = f.format(date);
    console.info(
      `Locale - "${locale}", name - "${languageName}", format result - "${text}"`,
    );
  });
})();
```

Tested in Linux Mint 21 Cinnamon (64-bit).

Output in Mozilla Firefox 106.0.3 (64-bit):

```
Locale - "kz", name - "kz", format result - "Sunday, March 22, 2020 at 3:00:00 PM East Kazakhstan Time"
Locale - "kk", name - "Kazakh", format result - "2020 ж. 22 наурыз, жексенбі, 15:00:00 Шығыс Қазақстан уақыты"
```

Output in Google Chrome 107.0.5304.87 (Official Build) (64-bit), Opera 92.0.4561.33, Electron 21.1.1 (launched using nodejs/npm):

```
Locale - "kz", name - "kz", format result - "Sunday, March 22, 2020 at 3:00:00 PM East Kazakhstan Time"
Locale - "kk", name - "Kazakh", format result - "Sunday, March 22, 2020 at 3:00:00 PM East Kazakhstan Time"
```

Output in Microsoft Edge 107.0.1418.28 (Official build) (64-bit):

```
Locale - "kz", name - "kz", format result - "Sunday, March 22, 2020 at 3:00:00 PM East Kazakhstan Time"
Locale - "kk", name - "Kazakh", format result - "2020 ж. 22 наурыз, жексенбі, 15:00:00 East Kazakhstan Time"
```

Format result is different in different browsers, but each browser cannot find name for language "kz" and recognizes locale "kk" as language "Kazakh".

Another test - in Java/Kotlin. I ran code using Adoptium JDK 18.0.2.1+1 (based on OpenJDK) and kotlin 1.7.20:

```kotlin
val kz = Locale("kz")
println(kz.getDisplayName(kz))

val kk = Locale("kk")
println(kk.getDisplayName(kk))
```

Output was:

```
kz - kz
kk - қазақ тілі
```

Same as in browsers - only locale "kk" has been recognized as "Kazakh".

So i think, language pack for kazakh must be saved in file "kk.js" instead of "kz.js".

*When country code ("kz" in this case) is used as language code - it is mistake. It is same as using "ca" (Canada) instead of "en" (English) or "fr" (French). And such mistake is widespread in software, developed in Kazakhstan, it is our tradition in some sort.*

**Comment by tolybay63** [comment]
Hello!
I think you are right. Thank you.
Please fix it if possible. I am a beginner in this area.
>Воскресенье, 6 ноября 2022, 16:30 +06:00 от Roman Obuhov ***@***.***>:
> 
> 
>Hello!
>May be name of language pack file must be "kk.js"? I saw language code "kk" at  https://en.wikipedia.org/wiki/Kazakh_language , in right panel, in "Language codes" section.
>Also I tried to test locales in few browsers by running next javascript code in browser console:
>(() => {
>
>  console.clear();
>
>  const date = new Date(2020, 2, 22, 15, 00, 00, 000);
>
>  const languageNames = new Intl.DisplayNames(['en'], { type: 'language' });
>
>  ['kz', 'kk'].forEach((locale) => {
>
>    const languageName = languageNames.of(locale);
>
>    const f = new Intl.DateTimeFormat(locale, { dateStyle: 'full', timeStyle: 'full' });
>
>    const text = f.format(date);
>
>    console.info(
>
>      `Locale - "${locale}", name - "${languageName}", format result - "${text}"`,
>
>    );
>
>  });
>
>})();
>Tested in Linux Mint 21 Cinnamon (64-bit).
>Output in Mozilla Firefox 106.0.3 (64-bit):
>Locale - "kz", name - "kz", format result - "Sunday, March 22, 2020 at 3:00:00 PM East Kazakhstan Time"
>
>Locale - "kk", name - "Kazakh", format result - "2020 ж. 22 наурыз, жексенбі, 15:00:00 Шығыс Қазақстан уақыты"
>
>
>Output in Google Chrome 107.0.5304.87 (Official Build) (64-bit), Opera 92.0.4561.33, Electron 21.1.1 (launched using nodejs/npm):
>Locale - "kz", name - "kz", format result - "Sunday, March 22, 2020 at 3:00:00 PM East Kazakhstan Time"
>
>Locale - "kk", name - "Kazakh", format result - "Sunday, March 22, 2020 at 3:00:00 PM East Kazakhstan Time"
>
>
>Output in Microsoft Edge 107.0.1418.28 (Official build) (64-bit):
>Locale - "kz", name - "kz", format result - "Sunday, March 22, 2020 at 3:00:00 PM East Kazakhstan Time"
>
>Locale - "kk", name - "Kazakh", format result - "2020 ж. 22 наурыз, жексенбі, 15:00:00 East Kazakhstan Time"
>
>
>Format result is different in different browsers, but each browser cannot find name for language "kz" and recognizes locale "kk" as language "Kazakh".
>Another test - in Java/Kotlin. I ran code using Adoptium JDK 18.0.2.1+1 (based on OpenJDK) and kotlin 1.7.20:
>val kz = Locale("kz")
>
>println(kz.getDisplayName(kz))
>
>
>
>val kk = Locale("kk")
>
>println(kk.getDisplayName(kk))
>Output was:
>kz - kz
>
>kk - қазақ тілі
>
>
>Same as in browsers - only locale "kk" has been recognized as "Kazakh".
>So i think, language pack for kazakh must be saved in file "kk.js" instead of "kz.js".
>When country code ("kz" in this case) is used as language code - it is mistake. It is same as using "ca" (Canada) instead of "en" (English) or "fr" (French). And such mistake is widespread in software, developed in Kazakhstan, it is our tradition in some sort.
>—
>Reply to this email directly,  view it on GitHub , or  unsubscribe .
>You are receiving this because you authored the thread. Message ID:  <quasarframework/quasar/repo-discussions/12663/comments/4067835 @ github . com> 
 
 
С уважением Толыбай Куанов.

**Reply by o-r-e** [reply]
Hello and thanks for reply!
I've created PR for such fix - https://github.com/quasarframework/quasar/pull/14985 ; will see if it is acceptable

With respect, с уважением, Роман
